tools/UpdateModules.ps1

```diff
@@ -26,34 +26,124 @@ function Create-ModulePsm1
   [CmdletBinding()]
   param(
     [string]$ModulePath,
-    [string]$TemplatePath
+    [string]$TemplatePath,
+    [bool]$AddDefaultParameters
   )
 
   PROCESS
   {
-	 $manifestDir = Get-Item -Path $ModulePath
-	 $moduleName = $manifestDir.Name + ".psd1"
-	 $manifestPath = Join-Path -Path $ModulePath -ChildPath $moduleName
-     $module = Test-ModuleManifest -Path $manifestPath
+     $manifestDir = Get-Item -Path $ModulePath
+     $moduleName = $manifestDir.Name + ".psd1"
+     $manifestPath = Join-Path -Path $ModulePath -ChildPath $moduleName
+     $file = Get-Item $manifestPath
+     Import-LocalizedData -BindingVariable ModuleMetadata -BaseDirectory $file.DirectoryName -FileName $file.Name
      $templateOutputPath = $manifestPath -replace ".psd1", ".psm1"
-     [string]$strict
-     [string]$loose
-     foreach ($mod in $module.RequiredModules)
+     [string]$importedModules
+     if ($ModuleMetadata.RequiredModules -ne $null)
      {
-        $strict += "  Import-Module " + $mod.Name + " -RequiredVersion " + [string]$mod.Version + "`r`n"
-        $loose += "  Import-Module " + $mod.Name + "`r`n"
+        foreach ($mod in $ModuleMetadata.RequiredModules)
+        {
+           if ($mod["ModuleVersion"])
+           {
+               $importedModules += Create-MinimumVersionEntry -ModuleName $mod["ModuleName"] -MinimumVersion $mod["ModuleVersion"]
+           }
+           elseif ($mod["RequiredVersion"])
+           {
+               $importedModules += "Import-Module " + $mod["ModuleName"] + " -RequiredVersion " + $mod["RequiredVersion"] + "`r`n"
+           }        
+        }
+     }
+
+     if ($ModuleMetadata.NestedModules -ne $null)
+     {
+         foreach ($dll in $ModuleMetadata.NestedModules)
+         {
+             $importedModules += "Import-Module (Join-Path -Path `$PSScriptRoot -ChildPath " + $dll.Substring(2) + ")`r`n"
+         }
      }
+
      $template = Get-Content -Path $TemplatePath
-     $template = $template -replace "%MODULE-NAME%", $module.Name
+     $template = $template -replace "%MODULE-NAME%", $file.BaseName
      $template = $template -replace "%DATE%", [string](Get-Date)
-     $template = $template -replace "%STRICT-DEPENDENCIES%", $strict
-     $template = $template -replace "%DEPENDENCIES%", $loose
+     $template = $template -replace "%IMPORTED-DEPENDENCIES%", $importedModules
+
+     if ($AddDefaultParameters) 
+     {
+        $nestedModules = $ModuleMetadata.NestedModules
+        $AllCmdlets = @()
+        $nestedModules | ForEach-Object {
+            $dllPath = Join-Path -Path $ModulePath -ChildPath $_
+            $Assembly = [Reflection.Assembly]::LoadFrom($dllPath)
+            $dllCmdlets = $Assembly.GetTypes() | Where-Object {$_.CustomAttributes.AttributeType.Name -contains "CmdletAttribute"}
+            $AllCmdlets += $dllCmdlets
+        }
+        
+        $FilteredCommands = @()
+        $AllCmdlets | ForEach-Object {
```

Comment:
This also feels like it could be in it's own function that returned a predicate, so that this function would basically be

$nestedModules | Get-CmdletType | Where-Object {Test-RequiredParameter -Cmdlet $_ -ParameterName "ResourceGroup"}

Reply:
done